### DISCUSSION TITLE
Week 11 Report

### DISCUSSION THREAD

**Discussion by mirak55** [SE-TINF22B6/CookPit#70] [General]
**Advancements:**

- UML-Diagrams
- sad.md file
- recipe overview
- input validation (recipe-maker)

**UML-Diagrams:**

In order to further explain the structure of our project we have created a component diagram.
![Component_Diagram](https://github.com/SE-TINF22B6/CookPit/assets/122603912/aaaec59b-ac66-4093-b7fb-8f3b97ac6966)

The provided component shows our projected divided in three main systems. 

**Frontend:**
The Frontend provides a interface for the user with the main features beeing the SearchEngine to search a recipe, the RecipeCreation via the OpenAI API, the Authentication to provide a login for the user to save his favorite recipes.

**Backend:**
In the Backend our API provides the communication between the user with his inputs and the OpenAI API which then turns the given ingredients into an recipe for the user.
In cooperation with the database the Authentication of a user via a login is provided, to give the user the option to save recipes.

**Database:**
The Database stores all Recipes added by any user. With the Search engine it is possible to search for specific recipes or to show a user his favorite recipes after he is logged in.

**sad.md file**

Some additional information on the component diagram and our archtecture is provided in our new created [sad ](https://github.com/SE-TINF22B6/CookPit/blob/main/sad.md)file.

**Recipe Overview:**

In order to also add some new features to our website, we are working on a page witxh is dedicated to show all or at least a set amount of recipes for the user to get some inspiration. 

![ShowAllRecipes](https://github.com/SE-TINF22B6/CookPit/assets/122603912/a5949c01-e8f2-406a-90ae-f3b112ef41a4)

**Recipe-Maker user validation**

We only want to create real, eatable and not deadly recipes. Therefore we now have a simple but effective user validation which will check each ingredient if its is a legitimate one or not. If not so, it will not be considered in the recipe making.

**Comment by BenjaminB109** [comment]
Dear CookPit Team,
I like the UML diagram very much. The descriptionof the individual components is clear and concise. The Recipe Overview is also very well done. I think it's nice that you can already see the most important data about the recipe in the overview.

**Comment by Nahlam4** [comment]
Dear Team Cookpit,
nice job on your component diagram, i just think you might need a connection of the Api-generated recipe to the database to save it. 
Also, I like the simplistic and straight-to-the-point Overview of your inspiration Recipes. 

Keep up the good work in the next Sprint! - Team DHBWhub

**Comment by ins305** [comment]
Hi Team CookPit,
Your component diagram together with your explanation make it clear and easily understandable how the elements of your project work together. 
Also, I like the design of your recipe overview page. It matches very well the overall look of your project.
Well done and keep up the good work!
